src/prog_algs/uncertain_data/unweighted_samples.py

```diff
@@ -15,13 +16,25 @@ def __init__(self, samples = []):
         """Initialize Unweighted Samples
 
         Args:
-            samples (array, optional): array of samples. Defaults to empty array.
+            samples (array, dict, optional): array of samples. Defaults to empty array. If dict, must be of the form of {key: [value, ...], ...}
         """
-        self.data = samples
+        if isinstance(samples, dict):
+            # Is in form of {key: [value, ...], ...}
+            # Convert to array of samples
+            if len(samples.keys()) == 0:
+                self.data = []  # is empty
+                return
+            n_samples = len(list(samples.values())[0])  # Number of samples
+            self.data = [{key: value[i] for key, value in samples.items()} for i in range(n_samples)]
+        elif isinstance(samples, Iterable):
+            # is in form of [{key: value, ...}, ...]
+            self.data = samples
+        else:
+            raise ValueError('Invalid input. Must be list or dict, was {}'.format(type(samples)))
 
     def sample(self, num_samples = 1):
         # Completely random resample
-        return UnweightedSamples(random.choice(self.data, num_samples))
+        return UnweightedSamples(random.choice(self.data, num_samples, replace = True))
```

Comment:
Looks good! One question though. What is the reasoning for sampling with replacement instead of without replacement? I don't really mind either way but am just curious.

Reply:
Mostly that sampling without replacement limits the number of samples to the size of the UnweightedSamples. Sampling with replacement allows us to sample larger sizes than the size of the unweighted samples object (with duplicates). 

This was the default behavior, this change just makes it explicit.